tsdb/index/tsi1/file_set.go

```diff
@@ -78,8 +78,9 @@ func (fs *FileSet) MustReplace(oldFiles []File, newFile File) *FileSet {
 
 	// Ensure all old files are contiguous.
 	for j := range oldFiles {
+		println("dbg/replace", len(fs.files), "//", i, j)
 		if fs.files[i+j] != oldFiles[j] {
-			panic("cannot replace non-contiguous files")
+			panic(fmt.Sprintf("cannot replace non-contiguous files: subset=%+v, fileset=%+v", Files(oldFiles).IDs(), Files(fs.files).IDs()))
```

Comment:
No, this is called in a goroutine started by `compact()`. It's an invariant of the code and should not fail. It's an extra check to ensure the index does not get corrupted.